tasks/deploy_qcow2.yml

```diff
@@ -0,0 +1,83 @@
+- name: Check if {{ miq_image_path }} is directory
+  stat:
+    path: "{{ miq_image_path }}"
+  register: image_path_st
+
+- name: Download the qcow image
+  get_url:
+    url: "{{ miq_qcow_url }}"
+    dest: "{{ image_path_st.stat.isdir | ternary( miq_image_path~'/'~miq_qcow_url.rpartition('/')[-1], miq_image_path) | regex_replace('//', '/') }}"
+    checksum: "{{ miq_image_checksum | default(omit) }}"
+  register: downloaded_file
+
+- name: Check file type
+  command: "/usr/bin/file {{ downloaded_file.dest | quote }}"
+  changed_when: false
+  register: filetype
+
+- name: Fail if image is not qcow
+  fail:
+    msg: "The downloaded file is not a valid QCOW file."
+  when: '"QCOW" not in filetype.stdout'
+
+- name: Calculate image size in GiB
+  set_fact:
+    miq_image_size_gib: "{{ filetype.stdout_lines[0].split()[5] | int // 2**30 }}"
+
+#
+# Find default disk size for miq disk:
+#
+- block:
+  - name: Extract integer from miq_vm_disk_size
+    set_fact:
+      miq_vm_disk_size_gib: "{{ miq_vm_disk_size | regex_replace('GiB$') }}"
+
+  - name: Fail if miq_vm_disk_size is less than  qcow size
+    fail:
+      msg: "Setting a disk size ({{ miq_vm_disk_size }}) lower than the image size ({{ miq_image_size_gib }}GiB) may result in disk corruption."
+    when: "miq_vm_disk_size_gib < miq_image_size_gib"
+  when: "miq_vm_disk_size is defined"
+
+#
+# Find default data storage domain for Miq disk:
+#
+- block:
+  - name: Fetch storages
+    ovirt_storage_domains_facts:
+      auth: "{{ ovirt_auth }}"
+
+  - name: Find data domain
```

Comment:
Is there a check to see if the data domain exists?

Reply:
If user specify wrong storage domain it will fail while creating the disk, with reasonbale error. If user don't specify any storage domain, we choose any data domain.